components/net/filemanager_thread.rs

```diff
@@ -85,6 +92,19 @@ impl FileManager {
         }).expect("Thread spawning failed");
     }
 
+    pub fn fetch_file(&self,
+                     sender: servo_channel::Sender<Data>,
+                     id: Uuid,
+                     check_url_validity: bool,
+                     origin: FileOrigin,
+                     response: &mut Response) {
+        let store = self.store.clone();
+        let mut r2 = response.clone();
```

Comment:
This cloned response is modified by try_fetch_file, but no other code observes the modifications and r2 is discarded.